### PR TITLE
Validate metadata for melanoma

### LINKED ISSUE #457: Validate metadata for melanoma
Add 4 new validators for `clinical` metadata:
* `mel_thick_mm`
  * values are coded as float, and can be in the CSV as anything like:
    * `1.23 mm`
    * `1.23mm`
    * `1.23`
    * `1`
  * this may be in the CSV as `mel_thick` alternatively
* `mel_class`
  * values:
    * `"melanoma in situ"`
    * `"invasive melanoma"`
    * `"recurrent/persistent melanoma, in situ"` (comma is optional in CSV)
    * `"recurrent/persistent melanoma, invasive"` (comma is optional in CSV)
    * `"melanoma nos"`
* `ulcer`
  * boolean value
* `mel_type`
  * values:
    * `"superficial spreading"` (alias `"ssm"`)
    * `"nodular"`
    * `"lentigo maligna melanoma"` (alias `"lmm"`)
    * `"acral lentiginous melanoma"` (alias `"alm"`)
    * `"NOS"`
    * `"spindle cell features"`
    * `null` (alias `"unknown"`)

All can still take CSV values of `"unknown"`, coded as `None`.

Finally, add an inter-field validation rule whereby if any of these are set to non-`None` values, then `diagnosis` must be exactly `"melanoma"`.

### LINKED ISSUE #478: Add metadata validation for "mitotic index"
`mel_mitotic_index` (also accept `mel_mit`)
Values are typically:
* `0/mm^2`
* `<1/mm^2`
* `1/mm^2`
* `2/mm^2`
* `3/mm^2`
* `4/mm^2`
* `>5/mm^2`
* `Unknown`

Since these are typically input as categorical values, we should discuss whether we want to coerce and store them as a float.

Like #457, this will only be valid if the `diagnosis` is `melanoma`.

### PR DESCRIPTION
**Pull request by msmolens**
Resolves #457 
Resolves #478

### CODE REVIEW

File: server/models/dataset_helpers/image_metadata.py [hunk 1]
```diff
@@ -567,6 +567,161 @@ def load(cls, value, acquisition, clinical, private):
...
+                    value = 'superficial spreading'
+                elif value == 'lmm':
+                    value = 'lentigo maligna melanoma'
+                elif value == 'alm':
+                    value = 'acral lentiginous melanoma'
+                cls._assertEnumerated(value, {
+                    'superficial spreading',
+                    'nodular',
+                    'lentigo maligna melanoma',
+                    'acral lentiginous melanoma',
```
[the remark this answers is not in the document]
Comment (kallooa) [reply server/models/dataset_helpers/image_metadata.py]: Sure just insert melanoma for all

File: server/models/dataset_helpers/image_metadata.py [hunk 2]
```diff
@@ -567,6 +567,161 @@ def load(cls, value, acquisition, clinical, private):
         acquisition[cls.name] = value
 
 
+class MelThickMmFieldParser(FieldParser):
+    name = 'mel_thick_mm'
+    allowedFields = {'mel_thick_mm', 'mel_thick'}
+    _formatRegex = re.compile(r"""
+        (.+?)    # Non-greedy
```
[the remark this answers is not in the document]
Comment (msmolens) [reply server/models/dataset_helpers/image_metadata.py]: https://docs.python.org/2/library/re.html#re.VERBOSE:

> Whitespace within the pattern is ignored

File: server/models/dataset_helpers/image_metadata.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (msmolens) [reply server/models/dataset_helpers/image_metadata.py]: To clarify, please note any corrections to the following list:
- `superficial spreading melanoma`
- `nodular melanoma`
- `lentigo maligna melanoma`
- `acral lentiginous melanoma`
- `melanoma NOS`
- `spindle cell features melanoma` (?)

[the remark this answers is not in the document]
Comment (kallooa) [reply server/models/dataset_helpers/image_metadata.py]: lgtm

**Pull request merged**